drop-boxes/register-raw-and-mzML/register-raw-and-mzml.py

```diff
@@ -0,0 +1,269 @@
+
+    mzmlDataSet.setMeasuredData(False)
+    mzmlDataSet.setSample(sample)
+    if time_stamp:
+        mzmlDataSet.setPropertyValue("Q_MEASUREMENT_START_DATE", time_stamp)
+    if not file_exists:
+        subprocess.call(["gzip", "-f", filepath])
+    zipped = filepath+".gz"
+    transaction.moveFile(zipped, mzmlDataSet)
+    return time_stamp
+
+'''Metadata extraction written by Chris. Support for batch upload (no metadata here) by Andreas'''
+def process(transaction):
+
+    context = transaction.getRegistrationContext().getPersistentMap()
+
+    # Get the incoming path of the transaction
+    incomingPath = transaction.getIncoming().getAbsolutePath()
+
+    key = context.get("RETRY_COUNT")
+    if (key == None):
+        key = 1
+
+    # Get the name of the incoming file
+    name = transaction.getIncoming().getName()
+    dataPath = os.path.join(incomingPath, name)
+
+    code = barcode_pattern.findall(name)[0]
+    if extract_barcode(code):
+        project = code[:5]
+        experiment = code[1:5]
+        parentCode = code[:10]
+    else:
+        raise ValueError("Invalid barcode: %s" % code)
+
+    search_service = transaction.getSearchService()
+    sc = SearchCriteria()
+    sc.addMatchClause(SearchCriteria.MatchClause.createAttributeMatch(SearchCriteria.MatchClauseAttribute.CODE, code))
+    found = search_service.searchForSamples(sc)
+    space = found[0].getSpace()
+    # find existing experiments
+    existingExperimentIDs = []
+    existingExperiments = search_service.listExperiments("/" + space + "/" + project)
+    for eexp in existingExperiments:
+        existingExperimentIDs.append(eexp.getExperimentIdentifier())
+
+    raw_path = None
+    mzml_path = None
+    openbis_format_code = None
+    for root, subFolders, files in os.walk(incomingPath):
+        if subFolders:
+            subFolder = subFolders[0]
+        for f in files:
+            stem, ext = os.path.splitext(f)
+            if ext.lower()=='.mzml':
+                if mzml_path:
+                    raise ValueError("More than one mzML found. Only one pair of raw data and mzML can be registered at a time")
+                mzml_path = f
+            if ext.lower() in VENDOR_FORMAT_EXTENSIONS:
+                if raw_path:
+                    raise ValueError("More than one raw file found. Only one pair of raw data and mzML can be registered at a time")
+                raw_path = f
+                openbis_format_code = VENDOR_FORMAT_EXTENSIONS[ext.lower()]
+    if not raw_path or not mzml_path:
+        raise ValueError("Did not find pair of raw data and mzML data - make sure both are contained in the folder")
+    # TODO allow complex barcodes in dropboxhandler so this can be changed to be more stable
+    prefixes = ms_prefix_pattern.findall(name)
+    if prefixes:
+        prefix = prefixes[0]
+        for p in prefixes:
+            if len(p) > prefix:
+                prefix = p
+        ms_code = prefix+code
+    else:
+        # workaround, keep?
```

Comment:
Remove or address question
```suggestion
        # workaround, keep?
```